pipeline_dp/dp_engine.py

```diff
@@ -649,3 +657,20 @@ def _check_data_extractors(data_extractors: pipeline_dp.DataExtractors):
         raise ValueError("data_extractors must be set to a DataExtractors")
     if not isinstance(data_extractors, pipeline_dp.DataExtractors):
         raise TypeError("data_extractors must be set to a DataExtractors")
+
+
+def _create_partition_selector(strategy: pipeline_dp.PartitionSelectionStrategy,
+                               budget: budget_accounting.MechanismSpec,
+                               max_partitions: int, pre_threshold: int):
+    if budget.standard_deviation_is_set:
+        assert strategy.is_thresholding
```

Comment:
can it happen and user of a library should address it? Then we should throw a more user-friendly message